### PR TITLE
Moves API types from FossaAPIV1 to Fossa.API.Types.

### PR DESCRIPTION
**Pull request by periodic**
# Overview

We had various types associated with the API scattered across two files.  This moves those types over to `Fossa.API.Types` where some similar types already exists.  This helps separate the types from the HTTP implementation which is important for future refactoring around the API.

## Acceptance criteria

No types are exported from `FossaAPIV1` that aren't related to HTTP.

## Testing plan

This is just moving types, so as long as it all compiles and tests work we should be fine.

I checked `fossa analyze` and `fossa test` on the local build with no problems.

## Checklist

- [x] I added tests for this PR's change (or confirmed tests are not viable).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] I updated `Changelog.md` if this change is externally facing. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] I updated `*schema.json` if I have made changes for `.fossa.yml`, `fossa-deps.{json, yaml, yml}`. You may also need to update these if you have added/removed new dependency (e.g. pip) or analysis target type (e.g. poetry).
- [x] I linked this PR to any referenced GitHub issues, if they exist.

### CODE REVIEW

**Comment by liftM** [comment]
Re: testing plan - I'm not gonna block this PR, but in general (even in Haskell) we should at least do a really quick manual test that the code paths being changed still behave properly. Just upload an analysis that exercises the code paths being changed. 

I've caught many surprising issues that way (e.g. weirdness with Template Haskell, functions doing different things when their inferred types changed, etc.), and it's a really cheap test that provides surprising value.

**Comment by periodic** [comment]
I'll run a quick analysis/upload with it then.

Are there any places in particular you worry about not being covered by the unit tests?

**Comment by liftM** [comment]
It looks like this PR touches `fossa analyze` and `fossa test`'s API routes. So run a quick `fossa analyze` and `fossa test` and if nothing crashes and the web app link works and the uploaded revision looks reasonable in the web app then it's probably fine. I don't think you even need to spin up a local instance for this - just use a test organization in production.

File: src/App/Fossa/FossaAPIV1.hs [hunk 1]
```diff
@@ -64,7 +55,7 @@ import Data.Text (Text)
 import Data.Text qualified as Text
 import Data.Word (Word8)
 import Effect.Logger
-import Fossa.API.Types (ApiOpts, ArchiveComponents, Issues, SignedURL, signedURL, useApiOpts)
+import Fossa.API.Types
```
Comment (scruffystuffs) [inline src/App/Fossa/FossaAPIV1.hs:58]: Don't use implicit imports:
https://github.com/fossas/fossa-cli/blob/master/docs/contributing/STYLE-GUIDE.md#dont-use-implicit-blanket-imports

We should also clean up the pre-existing implicit imports here, if you're using VSCode there's a code action for making imports explicit automatically (you'll have to run the formatter afterward).

Comment (scruffystuffs) [reply src/App/Fossa/FossaAPIV1.hs:58]: Also, we should probably remove the prelude import, it does nothing.

File: src/App/Fossa/API/BuildWait.hs [hunk 2]
```diff
@@ -24,7 +24,7 @@ import Control.Effect.StickyLogger (StickyLogger, logSticky')
 import Control.Timeout (Cancel, checkForCancel)
 import Data.Text (Text)
 import Effect.Logger (Logger, pretty, viaShow)
-import Fossa.API.Types (ApiOpts, Issues (..))
+import Fossa.API.Types
```
Comment (scruffystuffs) [inline src/App/Fossa/API/BuildWait.hs:27]: Don't use implicit imports:
https://github.com/fossas/fossa-cli/blob/master/docs/contributing/STYLE-GUIDE.md#dont-use-implicit-blanket-imports

We should also clean up the pre-existing implicit imports here, if you're using VSCode there's a code action for making imports explicit automatically (you'll have to run the formatter afterward).

**Review by scruffystuffs — APPROVED** [review]
Style guide nits, but otherwise LGTM

**Comment by periodic** [comment]
Just the link failing due to that one service being down.

**Pull request merged**